R/predict.R

```diff
@@ -72,6 +72,26 @@ predict.WrappedModel = function(object, task, newdata, subset = NULL, ...) {
     newdata = newdata[subset, , drop = FALSE]
   }
 
+
+  # set factor levels, present in test but missing in train, to NA
+  if (model$learner$fix.factors.prediction == TRUE &&
+      any(vcapply(newdata, function(x) class(x)) == "factor") &&
+      any(class(model$learner.model) == "lm" | class(model$learner.model) == "glmmPQL")) {
```

Comment:
Why only those particular models? If the fixing should be done only for models of a particular type, this should be a property of the learner that can be checked here.